content/blog/2019/02/2019-02-10-remoting-cli-removed.adoc

```diff
@@ -0,0 +1,40 @@
+---
+layout: post
+title: 'Remoting-based CLI removed from Jenkins'
+tags:
+- core
+- security
+- remoting
+author: jglick
+---
+
+Close to two years ago, we announced in
+link:../../2017/04/2017-04-11-new-cli/[New, safer CLI in 2.54]
+that the traditional “Remoting” operation mode of the Jenkins command-line interface
+was being deprecated for a variety of reasons, especially its very poor security record.
+Today support for this mode is finally being removed altogether,
+in both the server and bundled `jenkins-cli.jar` client.
+
+This change makes the code in Jenkins core related to the CLI considerably simpler and more maintainable.
+(There are still two _transports_—HTTP(S) and SSH—but they have similar capabilities and behavior.)
+It also reduces the “attack surface” the Jenkins security team must consider.
+Among other issues, a compromised _server_ could freely attack a developer’s laptop if `-remoting` were used.
+
+The
+link:../../../doc/upgrade-guide/2.46/[2.46.x upgrade guide]
+already urged administrators to disable Remoting mode on the server.
+Those Jenkins users who rely on the CLI for remote scripting (as opposed to the HTTP(S) REST APIs)
+would be affected only if they were still using the `-remoting` CLI flag,
+since the default has long been to use HTTP(S) mode.
+
+Most CLI features have long worked fine without `-remoting`,
+in some cases using slightly different syntax such as requiring shell redirects to access local files.
+As part of this change, some CLI commands, options, and option types in Jenkins core have been removed, other than `-remoting` itself:
+* The `login` and `logout` commands, and the `--username` and `--password` options.
```

Comment:
Does not render properly on GitHub, might need an extra empty line. Did you test this?

Reply:
> Did you test this?

`make run` failed with

```
make: *** No rule to make target 'node_modules/tether/.npmignore', needed by 'build/assets'.  Stop.
```

whatever that means. Will check rendering in GH at least.